app/models/miq_expression.rb

```diff
@@ -681,16 +681,25 @@ def _to_sql(exp, tz)
     when "not like"
       field = Field.parse(exp[operator]["field"])
       clause = field.does_not_match("%#{exp[operator]["value"]}%").to_sql
-    when "and", "or"
-      operands = exp[operator].collect do|operand|
-        o = _to_sql(operand, tz) if operand.present?
-        o.blank? ? nil : o
-      end.compact
-      if operands.length > 1
-        clause = "(" + operands.join(" #{self.class.normalize_sql_operator(operator)} ") + ")"
-      elsif operands.length == 1 # Operands may have been stripped out during pre-processing
-        clause = "(" + operands.first + ")"
+    when "and"
+      operands = exp[operator].each_with_object([]) do |operand, result|
+        next if operand.blank?
+        sql = _to_sql(operand, tz)
+        next if sql.blank?
+        result << sql
+      end
+      clause = Arel::Nodes::And.new(operands.collect { |o| Arel::Nodes::SqlLiteral.new(o) }).to_sql
+    when "or"
```

Comment:
Sure! The short answer is that Arel's public API doesn't do them in the same way. So where you can do `And.new([any, number, of, nodes])`, with `Or` you can only do `Or.new(left_side, right_side)`, so you have to compose them. @kbrock can you corroborate that?

The code that collects the sub-expressions is identical - I can try to collapse back into one if you think that's better. It happened that way more because I was going to do these one at a time, but then decided to stick them both in the same PR.